Proposal: Contributor Experience SIG (#2162)
open-telemetry/community: projects/contributor-experience.md

* Only implicit and unstructured recognition for contributors, which would help
  contributors to feel seen and recognized.
* No structure collaboration across SIGs, especially in the case of cross-cutting
  concerns (Security, Docs, End-User, ...)
* SIGs are short of contributors and require some help to advertise their need
  (blog post, social posts, etc) and to lower the barrier of entry into their
  sub-project.
* _add more_

Note that there are individuals within the project who try to tackle some of those
aspects as individuals and not in a team effort, notable:

* [Add skeleton for maintainer and contributor guidebooks](https://github.com/open-telemetry/community/pull/2051)
* Mentorship efforts, e.g. [Outreachy](https://cloud-native.slack.com/archives/C060GFUL0P6)
* Surveys by the End User SIG that aim towards contributor experience
* _add more_

Also related to this initiative is the [Check-In process of the Governance Committee](https://github.com/open-telemetry/community/blob/main/gc-check-ins.md)

### Goals, objectives, and requirements

The goal of this project is to establish a cross-cutting "SIG Contributor Experience",

> which is "responsible for improving the experience of those who upstream contribute to

Reviewer: Those who upstream contribute?

Author: This is a verbatim copy from the k8s contributor experience charter, I can reword it to "those who contribute upstream"

Reviewer: Alright, it sounded weird but it's still comprehensible

Author: Since it is a quote I will leave it as is

the OpenTelemetry project. [They] do this by creating, and maintaining programs and processes that promote community health and reduce project friction, while retiring those programs and processes that don't. Being conscientious of our contributor base is critical to scaling the project, growing the ecosystem, and helping the project succeed". (via [K8s Contributor Experience Special Interest Group Charter](https://github.com/kubernetes/community/blob/master/sig-contributor-experience/charter.md))

While this is "yet another SIG" initiative, we believe that **not** establishing
such a SIG is not an option: the problems outlined above will get bigger not
smaller and while the final ownership of the health of the project is with the GC
this requires a coordinated effort and more hands across the project to be addressed
successfully.

Note that similar to other cross-cutting SIGs (Docs, Security, End-User) this SIG will not exclusively own
the domain of contributor experience, but they take responsibility to drive efforts in close collaboration
with SIGs and Committees.

> ## Deliverables

Reviewer 2: Not sure if it needs explicit mention in the list of deliverables, but as I have discussed with some of you IMO an interesting area to explore here is to have more blog posts on the style of  "here's what's happening under the hood" or "we had this interesting technical problem which we fixed in this way" from SIGs. [The "Inside Rust" blog](https://blog.rust-lang.org/inside-rust/index.html) is a good example of the type of content that I think would be beneficial here.

There are other benefits, but in terms of Contributor Experience:
1. It can serve as recognition for contributors (we recognize you worked on this and you get to share with the community and the wider world
2. It can get more people interested and excited about development within OpenTelemetry (most content right now is focused on attracting more _users_ to OpenTelemetry, not so much developers)

Author: Great idea, I'll add that via https://github.com/open-telemetry/community/pull/2162/files#r1646316497

After foundation the SIG will collaborate with contributors of existing efforts
(see above) and work towards establishing programs and initiatives around contributor
experience, including

* Onboarding rituals for members, approvers, maintainers
* A contributor guidebook
* Mentorship opportunities

> * Contributor recognition programs (e.g. via [Credly](https://credly.com))

Author: ```suggestion
* Contributor recognition programs (e.g. via [Credly](https://credly.com) or with blog posts on the style of "here's what's happening under the hood"
* Blog posts and social media posts to highlight areas where help is needed
```

* Events targeting the Contributor Community
* Consume SIG Check-In reports from the GC and use this data to improve contributor experience

## Staffing / Help Wanted

GC/TC sponsors:

* [@jpkrohling](https://github.com/jpkrohling)
* [@svrnm](https://github.com/svrnm)
  
Maintainers, approvers, and contributors:

* [@svrnm](https://github.com/svrnm)
* [@jpkrohling](https://github.com/jpkrohling)

> 

Reviewer 2: I would be happy to help here (as a contributor at least for now, I can't promise to be able to invest a lot of time here, but at least I can review/provide feedback :))
```suggestion
* [@mx-psi](https://github.com/mx-psi)

```

## Meeting Times

> The SIG will meet every other week. Initial suggestion: Every other Monday, 6pm CET/10am PT when

Reviewer: We can certainly settle this later, but can we have this one earlier, especially if we have only EU/East Coast folks? Or have two alternating times. We could even then try to recruit more people in APAC.

Author: I just wanted to make a suggestion here to have something to discuss, but I agree

Reviewer 2: I would be interested in joining and having something earlier and more Europe-timezone-friendly would help in that

Author: I guess the same is true for @theletterf

Reviewer 3: I would ideally like to participate in this SIG, but can't if it's a lot earlier than 10am because I'm in Pacific Time and I personally don't do well before 8am.  I have good overlap later in my day and into the evening with some of APAC but not all of it - India time and Australia time are quite different.  It's okay if I can't be included though, I totally understand!

Author: @musingvirtual the time of the meeting is discussed and finalized when we have formed the SIG, if we can find a time that works for everyone great, if not, we all can participate in SIGs without attending meetings, and if this ever turns out to be an issue this SIG is a prime candidate for having 2 meetings to improve contributor experience ;-)

the domain of contributor experience, but they take responsibility to drive efforts in close collaboration
with SIGs and Committees.

## Deliverables

After foundation the SIG will collaborate with contributors of existing efforts
(see above) and work towards establishing programs and initiatives around contributor
experience, including

* Onboarding rituals for members, approvers, maintainers
* A contributor guidebook
* Mentorship opportunities
* Contributor recognition programs (e.g. via [Credly](https://credly.com) or with blog posts on the style of "here's what's happening under the hood"
* Blog posts and social media posts to highlight areas where help is needed
* Events targeting the Contributor Community
* Consume SIG Check-In reports from the GC and use this data to improve contributor experience

## Staffing / Help Wanted

GC/TC sponsors:

* [@jpkrohling](https://github.com/jpkrohling)
* [@svrnm](https://github.com/svrnm)
  
Maintainers, approvers, and contributors:

* [@svrnm](https://github.com/svrnm)
* [@jpkrohling](https://github.com/jpkrohling)
* [@mx-psi](https://github.com/mx-psi)

> 

Author: ```suggestion
* [theletterf](https://github.com/theletterf)
```

> * [@svrnm](https://github.com/svrnm)
> * [@jpkrohling](https://github.com/jpkrohling)
> * [@mx-psi](https://github.com/mx-psi)
> * [theletterf](https://github.com/theletterf)
> ## Meeting Times

Author: ```suggestion
* [@svrnm](https://github.com/svrnm)
* [@jpkrohling](https://github.com/jpkrohling)
* [@mx-psi](https://github.com/mx-psi)
* [theletterf](https://github.com/theletterf)

## Meeting Times
```

In this project proposal we suggest to form a cross-cutting special interest group
that takes responsibility for driving contributor experience projects and that
closely collaborates with all other SIGs and governing bodies to increase contributor
sustainability, velocity and happiness.

### Current challenges

Among others the challenges the OpenTelemetry project is facing in regards of
contributor experience are:

* No consistent way of moving contributors "up the ladder", e.g. from member to
  approver, from approver to maintainer etc, which makes those changes invisible
  to the larger project and opaque to project leadership on who is doing
  what.
* No clear onboarding path for new contributors, or contributors taking on new
  responsibilities within the project (triagers, approvers, maintainers, committee
  members), which leads to friction in understanding of the overall project goals,
  to misalignment in implementations and the feeling of contributors being remote
  to other parts of the project.
* No explicit contributor guidebook that helps contributors to answer common
  questions no matter where they are on the contributors' ladder.

> * Mentorship for new contributors or contributors stepping up into new roles to
>   make them part of the project and aware of.  

Reviewer 4: ```suggestion
* No mentorship available for new contributors or contributors stepping up into new roles to
  make them part of the project and aware of expectations.  
```
Does this update match the intended challenge statement?

> * Only implicit and unstructured recognition for contributors, which would help
>   contributors to feel seen and recognized.

Reviewer 4: ```suggestion
* Only implicit and unstructured recognition for contributors, instead of explicit
   recognition which would help contributors to feel seen and recognized.
```

> * No structure collaboration across SIGs, especially in the case of cross-cutting
>   concerns (Security, Docs, End-User, ...)

Reviewer 4: ```suggestion
* No structured collaboration across SIGs, especially in the case of cross-cutting
  concerns (Security, Docs, End-User, ...), leading to information silos and misalignment.
```

the domain of contributor experience, but they take responsibility to drive efforts in close collaboration
with SIGs and Committees.

## Deliverables

After foundation the SIG will collaborate with contributors of existing efforts
(see above) and work towards establishing programs and initiatives around contributor
experience, including

* Onboarding rituals for members, approvers, maintainers
* A contributor guidebook
* Mentorship opportunities
* Contributor recognition programs (e.g. via [Credly](https://credly.com) or with blog posts on the style of "here's what's happening under the hood"
* Blog posts and social media posts to highlight areas where help is needed
* Events targeting the Contributor Community
* Consume SIG Check-In reports from the GC and use this data to improve contributor experience

## Staffing / Help Wanted

GC/TC sponsors:

* [@jpkrohling](https://github.com/jpkrohling)
* [@svrnm](https://github.com/svrnm)
  
Maintainers, approvers, and contributors:

* [@svrnm](https://github.com/svrnm)
* [@jpkrohling](https://github.com/jpkrohling)
* [@mx-psi](https://github.com/mx-psi)
* [@theletterf](https://github.com/theletterf)

> 

Reviewer 4: ```suggestion
* [@JamieDanielson](https://github.com/JamieDanielson)

```
I would like to help with this project!

  questions no matter where they are on the contributors' ladder.
* No mentorship available for new contributors or contributors stepping up into new roles to
  make them part of the project and aware of expectations.  
* Only implicit and unstructured recognition for contributors, instead of explicit
   recognition which would help contributors to feel seen and recognized.
* No structured collaboration across SIGs, especially in the case of cross-cutting
  concerns (Security, Docs, End-User, ...), leading to information silos and misalignment.
* SIGs are short of contributors and require some help to advertise their need
  (blog post, social posts, etc) and to lower the barrier of entry into their
  sub-project.
* _add more_

Note that there are individuals within the project who try to tackle some of those
aspects as individuals and not in a team effort, notable:

* [Add skeleton for maintainer and contributor guidebooks](https://github.com/open-telemetry/community/pull/2051)
* Mentorship efforts, e.g. [Outreachy](https://cloud-native.slack.com/archives/C060GFUL0P6)
* Surveys by the End User SIG that aim towards contributor experience
* _add more_

> Also related to this initiative is the [Check-In process of the Governance Committee](https://github.com/open-telemetry/community/blob/main/gc-check-ins.md)

Reviewer 3: Another related resource is the CNCF's cross-cutting Contributor Experience SIG.  https://github.com/kubernetes/community/tree/master/sig-contributor-experience

Feels like one of the things this group might do is maintain a relationship with them and consult them for best practices and resources where that makes sense, as well as upstream our governance resources into their repos for other projects when that makes sense.

Reviewer 3: How do we feel all this fits with the current role of the "community manager"? Maybe that was an interim solution and we need a ContribEx SIG, not a single community manager, but if we're going to continue that role we should address the relationship. cc @austinlparker

Author: > Feels like one of the things this group might do is maintain a relationship with them and consult them for best practices and resources where that makes sense, as well as upstream our governance resources into their repos for other projects when that makes sense.

Ah, let me another item for that :-)

Good point around the Community Manager role, @austinlparker wdyt?

Reviewer: I'm not sure I agree with that. I think the Community Manager handles things like event planning, which I don't see as part of this SIG.

Reviewer 3: @jpkrohling - well, it's possible that the answer really is that ContribEx doesn't overlap with community management.

My sense based on what you've said and what I already think, is that the distinction is that ContribEx is focused on the internal contributor community, and the community manager is focused on how OpenTelemetry as a project shows up in developer communities external to OpenTelemetry.

But I think that's worth articulating clearly even if we're writing down the obvious, because "community manager" could be a reasonable job title for either of those focuses.

Reviewer: I think this can be a good thing to add to the charter. I'm marking this as resolved for now.

While this is "yet another SIG" initiative, we believe that **not** establishing
such a SIG is not an option: the problems outlined above will get bigger not
smaller and while the final ownership of the health of the project is with the GC
this requires a coordinated effort and more hands across the project to be addressed
successfully.

Note that similar to other cross-cutting SIGs (Docs, Security, End-User) this SIG will not exclusively own
the domain of contributor experience, but they take responsibility to drive efforts in close collaboration
with SIGs and Committees.

## Deliverables

After foundation the SIG will collaborate with contributors of existing efforts
(see above) and work towards establishing programs and initiatives around contributor
experience, including

* Onboarding rituals for members, approvers, maintainers
* A contributor guidebook
* Mentorship opportunities
* Contributor recognition programs (e.g. via [Credly](https://credly.com) or with blog posts on the style of "here's what's happening under the hood"
* Blog posts and social media posts to highlight areas where help is needed
* Events targeting the Contributor Community

> * Consume SIG Check-In reports from the GC and use this data to improve contributor experience

Reviewer 3: Do we think maintaining and/or enforcing a code of conduct should be a responsibility of this SIG, or does that stay with GC?

Reviewer 3: Does contributor diversity - both transparency reports and recruiting - feel like something that should be on this list?

Reviewer 3: If we're able to support this, it might be nice to add a second weekly maintainer meeting that's friendly to more time zones, and to re-examine together with GC how those meetings could best serve maintainers - I observe that not all SIGs send representatives.

Author: > Should maintaining and/or enforcing the code of conduct be a responsibility of this SIG?

The SIG can help with that, but this is something that remains with the @open-telemetry/governance-committee as a responsibility.

> Does contributor diversity feel like something that should be on this list?

Yes! 🙌 

> If we're able to support this, it might be nice to add a second weekly maintainer meeting that's friendly to more time zones

I have thought about this a few times as well. This is now one of the examples where such a SIG can bring a proposal to another SIG (or in that case to the group of maintainers/the GC) but it's not within their capability to execute that.

I will update the list with items trying to address your suggestions

> * Consume SIG Check-In reports from the GC and use this data to improve contributor experience

Author: ```suggestion
* Consume SIG Check-In reports from the GC and use this data to improve contributor experience
* Support the community of contributors to maintain the Code of Conduct (Note: this is a support function, the responsibility for the CoC and especially enforcing it lays with the Governance Committee)
* Implement or suggest measures to support contributor diversity
* Help SIGs to establish practices to improve contributor experience, e.g. having multiple meeting times to serve more timezones
```

## Staffing / Help Wanted

GC/TC sponsors:

* [@jpkrohling](https://github.com/jpkrohling)
* [@svrnm](https://github.com/svrnm)
  
Maintainers, approvers, and contributors:

* [@svrnm](https://github.com/svrnm)
* [@jpkrohling](https://github.com/jpkrohling)
* [@mx-psi](https://github.com/mx-psi)
* [@theletterf](https://github.com/theletterf)

> * [@JamieDanielson](https://github.com/JamieDanielson)

Author: ```suggestion
* [@musingvirtual](https://github.com/musingvirtual)
```

Reviewer 4: @svrnm Looks like I was accidentally removed from this 😅

Author: oh no, I am sorry! Let me fix this

## Meeting Times

> The SIG will meet every other week. Initial suggestion: Every other Monday, 6pm CET/10am PT when
> the Comms SIG is not meeting.

Author: ```suggestion
The SIG will meeting time will be discussed and finalized when this proposal has been merged. There will be a weekly meeting, if necessary to cater as many time zones as possible there will be bi-weekly meetings at alternating times.
```

Note that similar to other cross-cutting SIGs (Docs, Security, End-User) this SIG will not exclusively own
the domain of contributor experience, but they take responsibility to drive efforts in close collaboration
with SIGs and Committees.

## Deliverables

After foundation the SIG will collaborate with contributors of existing efforts
(see above) and work towards establishing programs and initiatives around contributor
experience, including

* Onboarding rituals for members, approvers, maintainers
* A contributor guidebook
* Mentorship opportunities
* Contributor recognition programs (e.g. via [Credly](https://credly.com) or with blog posts on the style of "here's what's happening under the hood"
* Blog posts and social media posts to highlight areas where help is needed
* Events targeting the Contributor Community
* Consume SIG Check-In reports from the GC and use this data to improve contributor experience
* Support the community of contributors to maintain the Code of Conduct (Note: this is a support function, the responsibility for the CoC and especially enforcing it lays with the Governance Committee)
* Implement or suggest measures to support contributor diversity

> * Help SIGs to establish practices to improve contributor experience, e.g. having multiple meeting times to serve more timezones

Author: ```suggestion
* Help SIGs to establish practices to improve contributor experience, e.g. having multiple meeting times to serve more timezones
* Maintain a relationship with other Contributor Experience SIGs (e.g. from K8s) and TAG Contributor Strategy, to consult them for best practices and resources where that makes sense, as well as upstream our governance resources into their repos for other projects when that makes sense.
```

  what.
* No clear onboarding path for new contributors, or contributors taking on new
  responsibilities within the project (triagers, approvers, maintainers, committee
  members), which leads to friction in understanding of the overall project goals,
  to misalignment in implementations and the feeling of contributors being remote
  to other parts of the project.
* No explicit contributor guidebook that helps contributors to answer common
  questions no matter where they are on the contributors' ladder.
* No mentorship available for new contributors or contributors stepping up into new roles to
  make them part of the project and aware of expectations.  
* Only implicit and unstructured recognition for contributors, instead of explicit
   recognition which would help contributors to feel seen and recognized.
* No structured collaboration across SIGs, especially in the case of cross-cutting
  concerns (Security, Docs, End-User, ...), leading to information silos and misalignment.
* SIGs are short of contributors and require some help to advertise their need
  (blog post, social posts, etc) and to lower the barrier of entry into their
  sub-project.

> * _add more_

Reviewer 5: This may not be the way to write it, but I'd like to be able to focus on this issue. and also not dismiss it as a problem of the people on that side for not showing up on occasions a secondary timezone meeting was attempted. The root of the issue is we need to refocus on building with conscious and active involvement in making other timezones work. We don't want to accidentally treat one side of the earth as folks who are relegated to watching the others lead, or force them to interrupt their sleep or family duties.

```suggestion
* implicit bias towards founding members timezones, such the majority of SIG meetings occurring at 11pm or midnight UTC+8.
```

Author: Thanks for adding this, as someone who is semi-affected by that as well, I fully agree. A layer that adds to the problem is that an answer people often get is "if you want your issue to make progress, please come to the SIG meeting". I understand why we say that and I said (and say) the same thing myself. In a broader sense, there is also an implicit bias that people share a work culture, e.g. they know periods of time where most people are off (e.g. last and first week of the year), they understand certain terminology and to be fluent enough in English to participate in a SIG meeting.

I tried to broaden your suggestion to include that:

```suggestion
* Implicit bias towards US (and EU) working hours, rituals and modes of operation. Majority of SIG meetings occurring from early PST onward.
```

Reviewer 5: thanks for improving my suggestion @svrnm

Reviewer 5: here's the related issue to this for those not following  https://github.com/open-telemetry/community/issues/2176
